Lib/fontTools/varLib/instancer.py

```diff
@@ -1099,6 +1115,223 @@ def pruningUnusedNames(varfont):
+    Finally, we remove duplicates and ensure Format 4 tables preside over
+    the other formats.
+
+    The updated nametable will conform to the R/I/B/BI naming model.
+    """
+    if "STAT" not in varfont:
+        raise ValueError("Cannot update name table since there is no STAT table.")
+    stat = varfont["STAT"]
+    fvar = varfont["fvar"]
+
+    # The updated name table must reflect the new 'zero origin' of the font.
+    # If a user is instantiating a partial font, we will populate the
+    # unpinned axes with their default axis values.
+    fvarDefaults = {a.axisTag: a.defaultValue for a in fvar.axes}
+    axisCoords = axisLimits
+    for axisTag, val in fvarDefaults.items():
+        if axisTag not in axisCoords:
+            axisCoords[axisTag] = val
+        elif isinstance(axisCoords[axisTag], tuple):
+            axisCoords[axisTag] = val
+
+    stat_new = deepcopy(stat).table
+    _instantiateSTAT(stat_new, axisCoords)
+    checkMissingAxisValues(stat_new, axisCoords)
+
+    axisValueTables = stat_new.AxisValueArray.AxisValue
+    # remove axis Value Tables which have Elidable_AXIS_VALUE_NAME flag set
+    # Axis Value which have this flag enabled won't be visible in
+    # application font menus.
+    axisValueTables = [
+        v for v in axisValueTables if v.Flags & ELIDABLE_AXIS_VALUE_NAME != 2
+    ]
+    stat_new.AxisValueArray.AxisValue = axisValueTables
+    axisValueTables = _sortedAxisValues(stat_new, axisCoords)
+    _updateNameRecords(varfont, axisValueTables)
+
+
+def checkMissingAxisValues(stat, axisCoords):
+    seen = set()
+    axisValueTables = stat.AxisValueArray.AxisValue
+    designAxes = stat.DesignAxisRecord.Axis
+    for val in axisValueTables:
+        if val.Format == 4:
+            for rec in val.AxisValueRecord:
+                axisTag = designAxes[rec.AxisIndex].AxisTag
+                seen.add(axisTag)
+        else:
+            axisTag = designAxes[val.AxisIndex].AxisTag
+            seen.add(axisTag)
+
+    missingAxes = set(axisCoords) - seen
+    if missingAxes:
+        missing = ", ".join(f"{i}={axisCoords[i]}" for i in missingAxes)
+        raise ValueError(f"Cannot find Axis Value Tables {missing}")
+
+
+def _sortedAxisValues(stat, axisCoords):
+    # Sort and remove duplicates ensuring that format 4 axis Value Tables
+    # are dominant
+    axisValueTables = stat.AxisValueArray.AxisValue
+    designAxes = stat.DesignAxisRecord.Axis
+    results = []
+    seenAxes = set()
+    # sort format 4 axes so the tables with the most AxisValueRecords
+    # are first
+    format4 = sorted(
+        [v for v in axisValueTables if v.Format == 4],
+        key=lambda v: len(v.AxisValueRecord),
+        reverse=True,
+    )
+    nonFormat4 = [v for v in axisValueTables if v not in format4]
+
+    for val in format4:
+        axisIndexes = set(r.AxisIndex for r in val.AxisValueRecord)
+        if bool(seenAxes & axisIndexes) == False:
```

Comment:
don't need bool...I think

Reply:
We need to keep bool since `bool(set() & set()) == False` returns True, yet `set() & set() == False` returns False.

Reply:
`if not seenAxes & axisIndices:` would be idiomatic.

Reply:
But if you insist on the `bool()`, then: `if not bool(seenAxes & axisIndices):` would be best.

Reply:
Thanks!